src/olympia/devhub/forms.py

```diff
@@ -1396,3 +1396,38 @@ def save(self):
                 AddonCategory(addon=self.addon, category_id=category.id).save()
         # Remove old, outdated categories cache on the model.
         del self.addon.all_categories
+
+
+class SitePermissionGeneratorForm(forms.Form):
+    origin = forms.URLField(
+        label=_('Origin'),
+        widget=forms.TextInput(attrs={'placeholder': 'https://example.com'}),
+    )
+    site_permissions = forms.MultipleChoiceField(
+        label=_('Permissions'), choices=(('midi-sysex', 'WebMIDI'),)
+    )
+
+    def clean_origin(self):
+        actual_value = str(self.data.get('origin'))
+        value = self.cleaned_data.get('origin')
+        # Note that URLField should already ensure it's an URL.
+        error_message = _(
+            'Origin should be made of a scheme (protocol), a hostname (domain) and an '
+            'optional port'
+        )
+        try:
+            parsed = urlparse(value)
+        except ValueError:
+            raise forms.ValidationError(error_message)
+        if (
+            not parsed.scheme
+            or parsed.scheme not in ('https', 'http')
+            or not parsed.netloc
+            or not actual_value.startswith(parsed.scheme)
```

Comment:
I don't understand this one. If we accept the value, we'll be using `cleaned_data`, will we not? Why are we concerned with checking `actual_value`?

Reply:
I'll clarify in a comment. Django's `URLField` accepts `foo` as actual value, translating it to `http://foo` in the cleaned value. So I have to check the actual value here.